app/code/Magento/Catalog/Model/CategoryList.php

```diff
@@ -49,20 +43,17 @@ class CategoryList implements CategoryListInterface
      * @param CollectionFactory $categoryCollectionFactory
      * @param JoinProcessorInterface $extensionAttributesJoinProcessor
      * @param CategorySearchResultsInterfaceFactory $categorySearchResultsFactory
-     * @param CategoryRepositoryInterface $categoryRepository
      * @param CollectionProcessorInterface $collectionProcessor
      */
     public function __construct(
         CollectionFactory $categoryCollectionFactory,
         JoinProcessorInterface $extensionAttributesJoinProcessor,
         CategorySearchResultsInterfaceFactory $categorySearchResultsFactory,
-        CategoryRepositoryInterface $categoryRepository,
```

Comment:
Please preserve the constructor argument for backward compatibility

Reply:
I completely forgot about that, that was pretty stupid.. Thanks, readded.